### DISCUSSION TITLE
Althold and Poshold throttle stick hover position indicator.

### DISCUSSION THREAD

**Discussion by Jetrell** [iNavFlight/inav#8933] [Ideas]
For `Althold` and `Poshold` to work correctly.. It requires `nav_mc_hover_thr` to be set, according to the crafts requirements.
And the throttle stick is required to be placed at mid stick, or _technically_ should hold the stick position Althold is enabled at... Otherwise the craft will not hold altitude, and will either climb or descend.
There is `pos_hold_deadband` which can help, by widening this position a bit.

Presently getting the throttle stick in the exact position required for stable altitude control, after an altitude adjustment is made, isn't always intuitive, and can only be assessed after some observation.. With many people asking on Discord and here, why their MC won't hold altitude.

Some pilots prefer to use `nav_use_midthr_for_althold`. and others do not.

The RC throttle obviously can not be locked, due to it being required to change the altitude holding setpoint,
So maybe having a System message appear on the OSD, when either `Althold `or `Poshold` mode is selected... Informing the user to set the throttle stick to the hold position, **after** a positive or negative altitude change is made by them.
Maybe something like **(CENTER THROTTLE STICK TO LOCK ALTITUDE)**. Then the message disappearing once the change is made.

Idea's from others ?

**Comment by breadoven** [comment]
This seems related to what I mentioned in https://github.com/iNavFlight/inav/pull/8896.

With `nav_use_midthr_for_althold `OFF the RC adjustment zero throttle datum for alt hold is set to the current throttle position.  If the stick is close to max or min throttle it adjusts the zero datum to the `pos_hold_deadband `+ 10 to ensure there's some control available, or if the throttle is fully low it sets the zero datum to mid stick. RC adjustment is then scaled off this zero datum. Never tried it with the stick close to min or max throttle but a 10 RC point margin doesn't seem much and would make adjusting altitude a bit tricky in one direction I'd have thought. This could be an issue for example if you cancel RTH with the throttle stick close to min, switching directly into Poshold. Descent control would be very sensitive in this case if I read the code right, bordering on using a switch going from hover to max auto descent rate with a tiny movement of the stick. One option that I think would be useful here is to be able to use `nav_mc_hover_thr `as the zero datum setting rather than just mid throttle. A lot of quads seem to hover well below mid throttle so this setting doesn't seem particularly useful and causes problems when switching into ACRO.

Maybe a flashing inverted OSD altitude value would be one way of indicating RC altitude adjustment is active.

**Reply by Jetrell** [reply]
>One option that I think would be useful here is to be able to use `nav_mc_hover_thr `as the zero datum setting rather than just mid throttle. A lot of quads seem to hover well below mid throttle so this setting doesn't seem particularly useful and causes problems when switching into ACRO.

That idea sounds good..  All my quads have their hover throttle below 1300uS..

The way it is currently with `nav_use_midthr_for_althold = OFF`,  work fine the first time the mode is activated. But once an altitude change is made by the throttle stick. Its a matter of moving the stick around, and watching, to see if its holding altitude. 

> Maybe a flashing inverted OSD altitude value would be one way of indicating RC altitude adjustment is active.

Yes. That would be good.. It would certainly bring it to the pilots attention, and give them a certainty that the stick is in the correct position.

**Reply by Jetrell** [reply]
@breadoven I'll give an example of what I did in the test I was running that day.
I switched to `Althold + Airmode + Heading hold`... But I was quickly approaching some trees. And wanted to be sure I'd make it over them.. So I had to increase the altitude. Then I had trouble finding the altitude hold neutral climb point again, with all the other stuff I was taking note of.

This is likely why so many people report this issue. Because a lot of new INAV MC users fly around in `Poshold`.  
I can imagine that would be confusing for them to work out why the quad is rising or sinking. Without being able to look at the throttle stick, when wearing goggles.

**Reply by Jetrell** [reply]
>A lot of quads seem to hover well below mid throttle so this setting doesn't seem particularly useful and causes problems when switching into ACRO.

A guy on Discord brought up this issue the other day too.

He wanted to find a way to stop the quads altitude jumping up or down. When switching from `Poshold` to `Angle` mode.. 
 I explained to him the difference between the actual throttle stick position required for a hover in Angle mode. And the `Poshold` stick activation point..  Which are going to be slightly different in most cases, and cause that change in altitude until the throttle stick is manually trimmed to get it right.
